Amazon S3 5.4.0 M4 - TLS Config (#435)
mulesoft/docs-connectors: modules/ROOT/pages/amazon/amazon-s3-connector-studio.adoc

To configure a connector in Anypoint Studio:

. Add the connector to your Studio project.
. Configure an input source for the connector.
. Configure the connector.

== Add the Connector to Your Project

Anypoint Studio provides two ways to add the connector to your Studio project: from the Exchange button in the Studio taskbar or from the Mule Palette view.

=== Add the Connector Using Exchange

. In Studio, create a Mule project.
. Click the Exchange icon *(X)* in the upper-left of the Studio task bar.
. In Exchange, click *Login* and supply your Anypoint Platform username and password.
. In Exchange, search for "s3".
. Select the connector and click *Add to project*.
. Follow the prompts to install the connector.

=== Add the Connector in Studio

. In Studio, create a Mule project.
. In the Mule Palette view, click *(X) Search in Exchange*.
. In *Add Modules to Project*, type "s3" in the search field.
. Click this connector's name in *Available modules*.
. Click *Add*.
. Click *Finish*.

== Configure an Input Source

> Configure an input source for the Amazon S3 connector such as a connector operation, *HTTP Connector*, or *scheduler*.

Reviewer: ```suggestion
Configure an input source for Amazon S3 Connector, such as a connector operation, HTTP connector, or scheduler.
```

You can put the following operations for the connector in the *Source* area of the Studio canvas:

> * *On Deleted Object* - Initiates access to your app when an Amazon S3 object is deleted.

Reviewer: GLOBAL FORMAT CHANGE, ALL LISTS, PER STYLE REF:

```suggestion
* *On Deleted Object*
Initiates access to your app when an Amazon S3 object is deleted.
```

> * *On New Object* - Initiates access to your app when an Amazon S3 object is created.

Reviewer: DITTO

```suggestion
* *On New Object*
Initiates access to your app when an Amazon S3 object is created.
```

> == Configure Amazon S3

Reviewer: ```suggestion
== Configure Amazon S3 Connector
```

|Access Key |Alphanumeric text string that uniquely identifies the user who owns the AWS account.
|Secret Key |Key that plays the role of a password.
|Try Default AWS Credentials Provider Chain |Indicates whether to use temporary credentials.

> |Region Endpoint |URL entry point for a web service.

Reviewer: Ditto previous comment about table column fragments and periods

|===
+

> You can either enter your credentials into the global configuration properties or reference a configuration placed in a file in the `src` folder of your project file that contains these values. For simpler maintenance and better reuse of your project, Mule recommends that you use a configuration file.

Reviewer: ```suggestion
You can either enter your credentials into the global configuration properties or reference a configuration placed in a file in the `src` folder of your project file that contains these values. For simpler maintenance and better reuse of your project, use a configuration file.
```

+
. If you are:
+

> * Using the default Amazon S3 storage, leave the default entries for *Proxy*, *Security*, and *Advanced* tabs.

Reviewer: . If you are using the default Amazon S3 storage, leave the default entries for *Proxy*, *Security*, and *Advanced* tabs.

> * Connecting to a different storage than the default AWS S3, specify its URL in the the *Advanced* tab's *S3 Compatible Storage URL* field .

Reviewer: ```suggestion
. If you are connecting to different storage than the default AWS S3, specify its URL in the *Advanced* tab's *S3 Compatible Storage URL* field.
```

. Click *OK* to save the global connector configurations.

////
== Example of Configuring S3-Compatible Storage

> The https://github.com/minio/[MinIO Project] is one example of Amazon S3 storage. If you configure this storage locally, you need to set the S3 Compatible Storage URL to `+http://127.0.0.1:9000+`.

Reviewer: ```suggestion
The https://github.com/minio/[MinIO Project] is one example of Amazon S3 storage. If you configure this storage locally, set the S3 Compatible Storage URL to `+http://127.0.0.1:9000+`.
```
